File: api/app/Models/Picture.php
<?php

namespace App\Models;

class Picture extends Model
{
    protected $fillable = [
        'name',
        'hash',
        'date',
        'size',
        'width',
        'height',
        'album_id',
    ];

    public static function getPathStatic($userId, $albumId, $name = null): string {
        return Album::getPathStatic($userId, $albumId) . "/" . $name ?? "";
    }
    public function getPath($userId): string {
        return $this->getPathStatic($userId, $this->album_id, $this->name);
    }

    public static function getPathThumbStatic($userId, $albumId, $pictureId = null, $orientation = '', $size = ''): string {
        return Album::getPathStatic($userId, $albumId) . "/thumbs/" . ($pictureId ? "$pictureId-$orientation$size.jpg" : '');
    }
    public function getPathThumb($userId, $orientation, $size): string {
        return $this->getPathThumbStatic($userId, $this->album_id, $this->id, $orientation, $size);
    }

    public function resolveRouteBinding($value, $field = null) {
        $album = request()->route('album') ?? 'NULL';
        if ($album instanceof Album) $album = $album->id;
        return $this
            ->where('album_id', $album)
            ->where($field ?? 'id', $value)
            ->firstOrFailCustom();
    }

    public function album() {
        return $this->belongsTo(Album::class);
    }
    public function tagPictures() {
        return $this->hasMany(TagPicture::class);
    }
    public function tags() {
        return $this->belongsToMany(Tag::class, 'tag_pictures');
    }
    public function complaints() {
        return $this->hasMany(Complaint::class);
    }
}

File: client/PicsyncClient/Models/Picture.cs
using System.Text.Json.Serialization;

namespace PicsyncClient.Models;

public class Picture
{
    [JsonPropertyName("id")] public ulong? Id { get; set; }
    [JsonPropertyName("name")] public string? Name { get; set; }
    [JsonPropertyName("hash")] public string? Hash { get; set; }
    [JsonPropertyName("size")] public ulong? Size { get; set; }
    [JsonPropertyName("date")] public DateTime? Date { get; set; }
    [JsonPropertyName("width")] public int? Width { get; set; }
    [JsonPropertyName("height")] public int? Height { get; set; }
    [JsonPropertyName("uploadedAt")] public DateTime? UploadedAt { get; set; }

    public Uri? Thumbnail { get; set; }
}